### PR TITLE
RELATED: RAIL-4105 Make the title editable

### PR DESCRIPTION
**Pull request by krejzd**
Make the title editable

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7729/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9594/ : FAILURE in 5m 15s

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7748/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9603/ : FAILURE in 5m 09s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7772/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9631/ : SUCCESS in 10m 03s

File: libs/sdk-ui-dashboard/src/presentation/topBar/title/RenderModeAwareTitle.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import { DefaultTitle } from "./DefaultTitle";
+import { EditableTitle } from "./EditableTitle";
+import { renderModeAware } from "../../componentDefinition";
+
+/**
+ * @internal
+ */
+export const RenderModeAwareTitle = renderModeAware({
+    view: DefaultTitle,
+    edit: EditableTitle,
```
Comment (kandl) [inline libs/sdk-ui-dashboard/src/presentation/topBar/title/RenderModeAwareTitle.tsx:12]: I think that this task is also about calling the relevant command, when title is changed. (`calls the RenameDashboard command on blur or Enter press`)

Comment (kandl) [reply libs/sdk-ui-dashboard/src/presentation/topBar/title/RenderModeAwareTitle.tsx:12]: Also, please check https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like (at least, you should mention the ticket number in the commit message)

Comment (krejzd) [reply libs/sdk-ui-dashboard/src/presentation/topBar/title/RenderModeAwareTitle.tsx:12]: > I think that this task is also about calling the relevant command, when title is changed. (`calls the RenameDashboard command on blur or Enter press` https://jira.intgdc.com/browse/RAIL-4105)

It is already linked and called here https://github.com/gooddata/gooddata-ui-sdk/blob/2823b00b889f2624366ed6140d4c38e568a7b1ca/libs/sdk-ui-dashboard/src/presentation/dashboard/DashboardHeader/DashboardHeader.tsx#L259

Comment (krejzd) [reply libs/sdk-ui-dashboard/src/presentation/topBar/title/RenderModeAwareTitle.tsx:12]: > Also, please check https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like (at least, you should mention the ticket number in the commit message)

I fixed that

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7782/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9639/ : SUCCESS in 13m 03s

File: libs/sdk-ui-dashboard/src/presentation/topBar/title/RenderModeAwareTitle.tsx (hunk 1 above)
Comment (kandl) [reply libs/sdk-ui-dashboard/src/presentation/topBar/title/RenderModeAwareTitle.tsx:12]: Oh, I missed that as it's not part of this PR. Thanks for the explanation.
Thank you.

**Review by kandl — APPROVED** [review]
LGTM

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2485

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2254/ : SUCCESS in 10m 49s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2543/ : SUCCESS in 18s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1366/ : SUCCESS in 16m 00s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9642/ : SUCCESS in 10m 08s

**Pull request merged**